File: tarefas-compartilhadas/src/main/java/br/edu/tarefascompartilhadas/model/ComentarioTarefa.java
package br.edu.tarefascompartilhadas.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class ComentarioTarefa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "tarefa_id", nullable = false)
    private Tarefa tarefa;

    @ManyToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @Column(nullable = false, length = 2000)
    private String texto;

    private LocalDateTime dataComentario;

    @PrePersist
    public void prePersist() {
        if (dataComentario == null) {
            dataComentario = LocalDateTime.now();
        }
    }
}

File: app_tarefas_compartilhadas/lib/models/comentario_tarefa.dart
import 'tarefa.dart';
import 'usuario.dart';

class ComentarioTarefa {
  final int? id;
  final Tarefa? tarefa;
  final Usuario? usuario;
  final String? texto;
  final String? dataComentario;

  ComentarioTarefa({
    this.id,
    this.tarefa,
    this.usuario,
    this.texto,
    this.dataComentario,
  });

  factory ComentarioTarefa.fromJson(Map<String, dynamic> json) {
    return ComentarioTarefa(
      id: json['id'],
      tarefa: json['tarefa'] != null
          ? Tarefa.fromJson(json['tarefa'])
          : null,
      usuario: json['usuario'] != null
          ? Usuario.fromJson(json['usuario'])
          : null,
      texto: json['texto'],
      dataComentario: json['dataComentario'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tarefa': tarefa?.toJson(),
      'usuario': usuario?.toJson(),
      'texto': texto,
      'dataComentario': dataComentario,
    };
  }
}